### PR TITLE
fix(chat): add default values in application dialog (Issue #2477)

### PR DESCRIPTION
**Pull request by Derikyan**
**Description:**

Need to add default value for entrypoint/name/version/icon fields

Issues:

- Issue #2477 

**UI changes**

Снимок экрана 2024-10-31 в 12 06 14
Снимок экрана 2024-10-31 в 12 06 53
Снимок экрана 2024-10-31 в 12 06 33
Снимок экрана 2024-10-31 в 12 06 44

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2490.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2490.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2490.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Derikyan** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/533140)
>Environment URL: https://chat-ai-dial-chat-pr-2490.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/types/common.ts [hunk 1]
```diff
@@ -115,6 +115,7 @@ export interface DropdownSelectorOption {
 export interface SelectOption<L, V> {
   label: L;
   value: V;
+  defaultEndpoint?: string;
```
Comment (IlyaBondar) [inline apps/chat/src/types/common.ts]:
```suggestion
  defaultValue?: string;
```

File: apps/chat/src/components/Common/Forms/DynamicFormFields.tsx [hunk 2]
```diff
@@ -79,7 +79,7 @@ export const DynamicFormFields = <
   const handleAdd = (option?: SelectOption<string, string>) => {
     append({
       label: option?.value ?? '',
-      value: '',
+      value: option?.defaultEndpoint ?? '',
```
Comment (IlyaBondar) [inline apps/chat/src/components/Common/Forms/DynamicFormFields.tsx]:
```suggestion
      value: option?.defaultValue ?? '',
```

File: apps/chat/src/components/Common/ApplicationWizard/form.ts [hunk 3]
```diff
@@ -282,10 +285,17 @@ const getToolsetStr = (config: QuickAppConfig) => {
...
+  models?: ShareEntity[],
+): FormData => ({
+  name:
+    app?.name ??
+    (models
+      ? getNextDefaultName(DEFAULT_APPLICATION_NAME, models, 0, true)
+      : ''),
   description: app ? getModelDescription(app) : '',
-  version: app?.version ?? '',
+  version: app?.version ?? '0.0.1',
```
Comment (IlyaBondar) [inline apps/chat/src/components/Common/ApplicationWizard/form.ts]:
```suggestion
  version: app?.version ?? DEFAULT_VERSION,
```

File: apps/chat/src/components/Common/ApplicationWizard/form.ts [hunk 4]
```diff
@@ -282,10 +285,17 @@ const getToolsetStr = (config: QuickAppConfig) => {
...
-  name: app?.name ?? '',
+export const getDefaultValues = (
+  app?: CustomApplicationModel,
+  models?: ShareEntity[],
+): FormData => ({
+  name:
+    app?.name ??
+    (models
+      ? getNextDefaultName(DEFAULT_APPLICATION_NAME, models, 0, true)
+      : ''),
```
Comment (IlyaBondar) [inline apps/chat/src/components/Common/ApplicationWizard/form.ts]:
```suggestion
   getNextDefaultName(DEFAULT_APPLICATION_NAME, models ?? [], 0, true),
```

File: apps/chat/src/components/Common/ApplicationWizard/CodeAppView/CodeAppView.tsx [hunk 5]
```diff
@@ -85,6 +95,14 @@ export const CodeAppView: React.FC<ViewProps> = ({
 
   const files = useAppSelector(FilesSelectors.selectFiles);
   const topics = useAppSelector(SettingsSelectors.selectTopics);
+  const models = useAppSelector(ModelsSelectors.selectModels);
+  const installedModelIds = useAppSelector(
+    ModelsSelectors.selectInstalledModelIds,
+  );
+
+  const filteredModels = models
+    .filter((model) => installedModelIds.has(model.reference))
```
Comment (IlyaBondar) [inline apps/chat/src/components/Common/ApplicationWizard/CodeAppView/CodeAppView.tsx]: We should check through any models, not only installed

File: apps/chat/src/components/Common/ApplicationWizard/CustomAppView.tsx [hunk 6]
```diff
@@ -55,6 +56,14 @@ export const CustomAppView: React.FC<ViewProps> = ({
 
   const files = useAppSelector(FilesSelectors.selectFiles);
   const topics = useAppSelector(SettingsSelectors.selectTopics);
+  const models = useAppSelector(ModelsSelectors.selectModels);
+  const installedModelIds = useAppSelector(
+    ModelsSelectors.selectInstalledModelIds,
+  );
+
+  const filteredModels = models
+    .filter((model) => installedModelIds.has(model.reference))
```
Comment (IlyaBondar) [inline apps/chat/src/components/Common/ApplicationWizard/CustomAppView.tsx]: We should check through any models, not only installed

File: apps/chat/src/components/Common/ApplicationWizard/QuickAppView.tsx [hunk 7]
```diff
@@ -56,6 +57,14 @@ export const QuickAppView: React.FC<ViewProps> = ({
   const files = useAppSelector(FilesSelectors.selectFiles);
   const theme = useAppSelector(UISelectors.selectThemeState);
   const topics = useAppSelector(SettingsSelectors.selectTopics);
+  const models = useAppSelector(ModelsSelectors.selectModels);
+  const installedModelIds = useAppSelector(
+    ModelsSelectors.selectInstalledModelIds,
+  );
+
+  const filteredModels = models
+    .filter((model) => installedModelIds.has(model.reference))
```
Comment (IlyaBondar) [inline apps/chat/src/components/Common/ApplicationWizard/QuickAppView.tsx]: We should check through any models, not only installed

**Comment by Derikyan** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2490.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**